File: Backend/src/main/java/com/meltic/gmao/model/MetricConfig.java
package com.meltic.gmao.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
@Table(name = "metric_configs")
public class MetricConfig {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombreMetrica; // ej: "temperatura", "humedad", "presion"
    private String unidadSeleccionada; // ej: "°C", "°F", "Bar", "PSI"
    
    private Double limiteMB; // Muy Bajo
    private Double limiteB;  // Bajo
    private Double limiteA;  // Alto
    private Double limiteMA; // Muy Alto
    
    private boolean habilitado = true;

    @ManyToOne
    @JoinColumn(name = "maquina_id")
    @JsonIgnore
    private Maquina maquina;

    public MetricConfig() {}

    public MetricConfig(String nombreMetrica, String unidadSeleccionada, Double mb, Double b, Double a, Double ma) {
        this.nombreMetrica = nombreMetrica;
        this.unidadSeleccionada = unidadSeleccionada;
        this.limiteMB = mb;
        this.limiteB = b;
        this.limiteA = a;
        this.limiteMA = ma;
    }

    // Getters y Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getNombreMetrica() { return nombreMetrica; }
    public void setNombreMetrica(String nombreMetrica) { this.nombreMetrica = nombreMetrica; }

    public String getUnidadSeleccionada() { return unidadSeleccionada; }
    public void setUnidadSeleccionada(String unidadSeleccionada) { this.unidadSeleccionada = unidadSeleccionada; }

    public Double getLimiteMB() { return limiteMB; }
    public void setLimiteMB(Double limiteMB) { this.limiteMB = limiteMB; }

    public Double getLimiteB() { return limiteB; }
    public void setLimiteB(Double limiteB) { this.limiteB = limiteB; }

    public Double getLimiteA() { return limiteA; }
    public void setLimiteA(Double limiteA) { this.limiteA = limiteA; }

    public Double getLimiteMA() { return limiteMA; }
    public void setLimiteMA(Double limiteMA) { this.limiteMA = limiteMA; }

    public boolean isHabilitado() { return habilitado; }
    public void setHabilitado(boolean habilitado) { this.habilitado = habilitado; }

    public Maquina getMaquina() { return maquina; }
    public void setMaquina(Maquina maquina) { this.maquina = maquina; }
}

File: Frontend/meltic_gmao_app/lib/models/metric_config.dart
class MetricConfig {
  final int? id;
  final String nombreMetrica;
  String unidadSeleccionada;
  double? limiteMB;
  double? limiteB;
  double? limiteA;
  double? limiteMA;
  bool habilitado;

  MetricConfig({
    this.id,
    required this.nombreMetrica,
    required this.unidadSeleccionada,
    this.limiteMB,
    this.limiteB,
    this.limiteA,
    this.limiteMA,
    this.habilitado = true,
  });

  factory MetricConfig.fromJson(Map<String, dynamic> json) {
    return MetricConfig(
      id: json['id'],
      nombreMetrica: json['nombreMetrica'],
      unidadSeleccionada: json['unidadSeleccionada'] ?? '',
      limiteMB: json['limiteMB']?.toDouble(),
      limiteB: json['limiteB']?.toDouble(),
      limiteA: json['limiteA']?.toDouble(),
      limiteMA: json['limiteMA']?.toDouble(),
      habilitado: json['habilitado'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombreMetrica': nombreMetrica,
      'unidadSeleccionada': unidadSeleccionada,
      'limiteMB': limiteMB,
      'limiteB': limiteB,
      'limiteA': limiteA,
      'limiteMA': limiteMA,
      'habilitado': habilitado,
    };
  }

  MetricConfig copyWith({
    double? limiteMB,
    double? limiteB,
    double? limiteA,
    double? limiteMA,
    String? unidadSeleccionada,
    bool? habilitado,
  }) {
    return MetricConfig(
      id: id,
      nombreMetrica: nombreMetrica,
      unidadSeleccionada: unidadSeleccionada ?? this.unidadSeleccionada,
      limiteMB: limiteMB ?? this.limiteMB,
      limiteB: limiteB ?? this.limiteB,
      limiteA: limiteA ?? this.limiteA,
      limiteMA: limiteMA ?? this.limiteMA,
      habilitado: habilitado ?? this.habilitado,
    );
  }
}
